Source/EkoActionSheetViewController.m

```diff
@@ -148,62 +152,118 @@ - (void)configureView
+        }
+    }
 }
 
 - (CGFloat)visibleViewHeight
 {
     CGFloat headerHeight = kEkoActionSheetHeaderViewHeight;
-    CGFloat cellHeight = kEkoActionSheetCellHeight;
-    NSUInteger cellCount = self.dataSource.numberOfActions;
-    CGFloat paddingBottom = 0;
-    if ([self.dataSource respondsToSelector:@selector(actionSheetPaddingBottom)]) {
-        paddingBottom = [self.dataSource actionSheetPaddingBottom];
+    CGFloat rowHeight = kEkoActionSheetCellHeight;
+    
+    CGFloat tableHeight = 0;
+    for (EkoActionSheetItem *item in self.items) {
+        if (item.heightBlock) {
+            tableHeight += item.heightBlock();
+        } else {
+            tableHeight += rowHeight;
+        }
     }
     
-    return ceilf(headerHeight + cellHeight * cellCount + paddingBottom);
+    return ceilf(headerHeight + tableHeight);
 }
 
-@end
+#pragma mark - Blur View
 
-@implementation EkoActionSheetViewController (Actions)
+- (FXBlurView *)blurView
+{
+    if (!_blurView)
+    {
+        _blurView = [[FXBlurView alloc] initWithFrame:self.view.bounds];
+        _blurView.tintColor = [UIColor clearColor];
+        _blurView.backgroundColor = [UIColor clearColor];
+        _blurView.dynamic = NO;
+        _blurView.blurRadius = 0.0f;
+        _blurView.blurEnabled = YES;
+        [_blurView setBlurEnabled:YES];
+        _blurView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+        
+        UIView *tintView = [[UIView alloc] initWithFrame:_blurView.bounds];
+        tintView.backgroundColor = [UIColor blackColor];
+        tintView.alpha = kEkoActionSheetTintViewAlpha;
+        tintView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+        [self.blurView addSubview:tintView];
+    }
+    return _blurView;
+}
 
-#pragma mark - UITapGesture action
+- (void)showBlurViewOnView:(UIView*)view animationDuration:(CGFloat)animationDuration
+{
+    self.blurView.alpha = 0.0f;
+    [view addSubview:self.blurView];
+    
+    [UIView animateWithDuration:animationDuration animations:^{
+        self.blurView.alpha = 1.0f;
+    } completion:nil];
+}
 
-- (IBAction)dismissFilteringInterface:(id)sender {
-    [self cancelButtonDidTouched:self.cancelButton];
+- (void)removeBlurViewWithAnimationDuration:(CGFloat)animationDuration
+{
+    [UIView animateWithDuration:animationDuration animations:^{
+        self.blurView.alpha = 0.0f;
+    } completion:^(BOOL finished) {
+        [self.blurView removeFromSuperview];
+    }];
 }
 
-- (void)cancelButtonDidTouched:(id)sender
+#pragma mark - Actions
+
+- (void)cancelItemActionHandler
 {
-    if ([self.delegate respondsToSelector:@selector(didTouchedCancelButton)]) {
-        [self.delegate didTouchedCancelButton];
+    if (self.cancelItem.handler) {
+        __weak typeof(EkoActionSheetItem) *weakItem = self.applyItem;
+        self.cancelItem.handler(weakItem, self);
```

Comment:
yeah, and one more reason to have reference to actionsheet viewcontroller instances, because some root viewcontrollers logic could require animated or non-animated dismissal of an actionsheet.